### ISSUE TITLE
Client certificate credentials require non-empty password

### ISSUE THREAD

**Issue by gt-dbechtold** [GrammaTech/jenkins-codesonar-plugin#13]
If you specify a PKCS#12 client certificate credential for hub authentication, the plugin will ignore it if the password associated with the credential is empty.

**Comment by gt-dbechtold** [comment]
Error looks like:

```
13:45:17  ERROR: [CodeSonar] failed to authenticate. 
13:45:17  IOException: Received fatal alert: handshake_failure
13:45:17  Stack Trace: javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
...
13:45:17  	at org.apache.http.client.fluent.Executor.execute(Executor.java:262)
13:45:17  	at org.jenkinsci.plugins.codesonar.services.HttpService.execute(HttpService.java:171)
13:45:17  	at org.jenkinsci.plugins.codesonar.services.AuthenticationService.authenticate701(AuthenticationService.java:107)
13:45:17  	at org.jenkinsci.plugins.codesonar.services.AuthenticationService.authenticate(AuthenticationService.java:40)
13:45:17  	at org.jenkinsci.plugins.codesonar.CodeSonarPublisher.authenticate(CodeSonarPublisher.java:458)
13:45:17  	at org.jenkinsci.plugins.codesonar.CodeSonarPublisher.perform(CodeSonarPublisher.java:368)
...
```

**Comment by aseno-grammatech** [comment]
Addressed in PR #16

**Comment by gt-dbechtold** [comment]
We investigated this and implemented a work-around.  Unfortunately, I don't remember all of the details. I think we found that Jenkins would not permit us to read the private key for a certificate credentials object that is not password protected.  In order to allow users to specify a passwordless PKCS 12 file, we instead allowed users to specify their passwordless file as a "Secret file" type, rather than as  "Certificate" type.
